subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/suite/filter/TestSignaturesFilter.kt

```diff
@@ -0,0 +1,42 @@
+package com.avito.instrumentation.suite.filter
+
+internal fun TestsFilter.Test.matched(signatures: Set<TestsFilter.Signatures.TestSignature>): Boolean {
+    return signatures.any { signature ->
+        name.startsWith(signature.name) && (signature.deviceName?.equals(
+            deviceName.name
+        ) ?: true)
+    }
+}
+
+internal class IncludeByTestSignaturesFilter(
+    private val source: TestsFilter.Signatures.Source,
```

Comment:
I've merged filter logic by test signature. Signature means testName + deviceName that's how we saved test runs in a report-viewer.
And now we can add signature filter with 4 sources: Report, PreviousRun, Impact, Code